matrix/buffer.py

```diff
@@ -1149,6 +1149,12 @@ def update_buffer_name(self):
                              else "#" + room_name)
             elif self.room.canonical_alias:
                 room_name = self.room.canonical_alias
+
+                W.buffer_set(
+                    self.weechat_buffer._ptr,
+                    "localvar_set_canonical_alias",
+                    room.canonical_alias
```

Comment:
Room isn't defined in this scope. It would be nice if you test if this works as expected since I don't see how this wouldn't produce a traceback.